### ISSUE TITLE
events in contracts

### ISSUE THREAD

**Issue by dakom** [Lay3rLabs/avs-toolkit#9]
Currently, the contracts add attributes in response, which get added to a `wasm` event.

This is fine for use-cases where we know for certain there's just one event being emitted in a flow, but it's not very scalable and runs a risk of having attributes with the same key meant for different events being conflated

Additionally, the attributes are plain strings and need to be kept in sync with client/test code which is a bit error-prone.

Suggestion for an event trait (or some system) which has the following properties

1. Strongly typed
2. impls `TryFrom<cosmwasm_std::Event>`
3. impls `Into<cosmwasm_std::Event>`
4. has a `NAME` const or `name()` getter... something to get the event name

In other words, given an event that satisfies these properties (perhaps a trait), we can do the following:

1. Add it to a `Response`
2. find it by name in a list of events (e.g. in a transaction response)
3. try to convert an event matching that name into the strongly typed struct
